web/src/lib/utils/slideshow-history.ts

```diff
@@ -0,0 +1,40 @@
+export class SlideshowHistory {
+  private slideshowHistory: string[] = [];
+  private slideshowHistoryIndex = 0;
+
+  constructor(private onNavigate: (assetId: string) => void) {}
+
+  reset(assetId: string) {
+    this.slideshowHistory = [assetId];
+    this.slideshowHistoryIndex = 0;
+  }
+
+  append(assetId: string) {
+    this.slideshowHistory.push(assetId);
+
+    // If we were at the end of the slideshow history, move the index to the new end
+    if (this.slideshowHistoryIndex === this.slideshowHistory.length - 2) {
+      this.slideshowHistoryIndex++;
+    }
+  }
+
+  navigateForward(): boolean {
```

Comment:
Going along the lines of the playlist concept, I think it would be more appropriate to name this next and previous. What do you think?